### PR TITLE
[2/2] Migrate from megaboom - enable usage in external repositories

### PR DESCRIPTION
**Pull request by lpawelcz**
**Please do not merge before https://github.com/Pinata-Consulting/bazel-orfs/pull/1** 
This is part [2/2] of PR chain which enables the usage of `bazel-orfs` in other repositories.
Please note that this PR is based on changes from PR https://github.com/Pinata-Consulting/bazel-orfs/pull/1. It is expected that you will see commits from both branches in diff view. Please look only at changes starting from https://github.com/Pinata-Consulting/bazel-orfs/commit/b657e471ff794c503060d9dca8af8a28e62b7e4f
This PR will be rebased onto newest main once https://github.com/Pinata-Consulting/bazel-orfs/pull/1 is merged and then converted from draft to regular PR.

This PR modifies files copied in https://github.com/Pinata-Consulting/bazel-orfs/pull/1 to make `build_openroad` macro usable in external repositories, like e.g.: [megaboom](https://github.com/The-OpenROAD-Project/megaboom) .

Required changes include:
* introducing new `entrypoint` argument to `build_openroad` which is a path to Makefile which includes `config.mk`. This entrypoint Makefile should be present in the external repository that uses `bazel-orfs`. It is a workaround for issues created by running make targets in `bazel-orfs` repository inside bazel sandbox in context of the external repository (e.g. megaboom)
* wrapping hardcoded paths to dependencies from `bazel-orfs` repository (e.g. `orfs` script, `*.mk` files) into `Label()` so that those paths would be processed in the context of `bazel-orfs` repo. Required to provide correct paths to those dependencies when using `bazel-orfs` in external repos.
* introducing `BAZEL_ORFS` and `MOCK_AREA_TCL` environment variables for passing down correct paths to `*.mk` files and `mock_area.tcl` script. `BAZEL_ORFS` can also be used in `entrypoint.mk` to provide correct include path to `config.mk`.

### CODE REVIEW

File: config.mk [hunk 1]
```diff
@@ -4,14 +4,14 @@ export PLATFORM=asap7
...
--include synth-bazel.mk
--include synth_sdc-bazel.mk
--include floorplan-bazel.mk
--include place-bazel.mk
--include cts-bazel.mk
--include grt-bazel.mk
--include route-bazel.mk
--include final-bazel.mk
--include generate_abstract-bazel.mk
+-include $(BAZEL_ORFS)/clock_period-bazel.mk
```
[the remark this answers is not in the document]
Comment (lpawelcz) [reply config.mk:7]: True, I will provide the documentation in README

[the remark this answers is not in the document]
Comment (lpawelcz) [reply config.mk:7]: Updated README with some notes on the implementation.

**Comment by lpawelcz** [comment]
Reworked `_orfs` targets and renamed those to `_local_runner` targets.
Currently the usage looks like this:
```
bazel build L1MetadataArray_test_floorplan_local_runner
cd bazel-bin && chmod -R +w . && cd ..
./bazel-bin/L1MetadataArray_test_floorplan_local_runner do-floorplan
```

There is no need to build `L1MetadataArray_test_floorplan` dependencies in a separate build as those dependencies are now connected to `_local_runner_script` targets.

**Comment by oharboe** [comment]
merged

**Pull request closed without merging**